QuickFIXn/Fields/Converters/DateTimeConverter.cs

```diff
@@ -9,195 +11,685 @@ namespace QuickFix.Fields.Converters;
+    /// is <see langword="true"/>, the value will be in the format "HH:mm:ss.fff". Otherwise, the value
+    /// will be in the format "HH:mm:ss".
+    /// </returns>
+    public static string ToFIXTimeOnly(DateTime dt, bool includeMilliseconds)
+        => ToFIXTimeOnly(TimeOnly.FromDateTime(dt), includeMilliseconds);
+
+    /// <summary>
+    /// Converts the specified <see cref="TimeOnly"/> to a <see cref="string"/>.
+    /// </summary>
+    /// <param name="time">The value to convert.</param>
+    /// <param name="precision">The level of precision with which to format the fractional seconds component of <paramref name="time"/>.</param>
+    /// <returns>
+    /// A value representing <paramref name="time"/>. The value will begin in the format "yyyyMMdd-HH:mm:ss"
+    /// and end in fractional seconds whose precision is determined by <paramref name="precision"/>.
+    /// </returns>
+    /// <exception cref="ArgumentOutOfRangeException"><paramref name="precision"/> is an invalid value.</exception>
+    public static string ToFIXTimeOnly(TimeOnly time, TimeStampPrecision precision)
+    {
+        return precision switch
+        {
+            TimeStampPrecision.Second => time.ToString("HH:mm:ss", CultureInfo.InvariantCulture),
+            TimeStampPrecision.Millisecond => time.ToString("HH:mm:ss.fff", CultureInfo.InvariantCulture),
+            TimeStampPrecision.Microsecond => time.ToString("HH:mm:ss.ffffff", CultureInfo.InvariantCulture),
+            TimeStampPrecision.Nanosecond => $"{time:HH:mm:ss}.{SubsecondAsNanoseconds(time):000000000}",
+            _ => throw new ArgumentOutOfRangeException(nameof(precision)),
+        };
+    }
+
+    /// <summary>
+    /// Converts the time component of the specified <see cref="DateTime"/> to a <see cref="string"/>.
+    /// </summary>
+    /// <param name="dt">The value to convert.</param>
+    /// <param name="precision">The level of precision with which to format the fractional seconds component of <paramref name="dt"/>.</param>
+    /// <returns>
+    /// A value representing the time component of <paramref name="dt"/>. The value will begin in the format "yyyyMMdd-HH:mm:ss"
+    /// and end in fractional seconds whose precision is determined by <paramref name="precision"/>.
+    /// </returns>
+    /// <exception cref="ArgumentOutOfRangeException"><paramref name="precision"/> is an invalid value.</exception>
     public static string ToFIXTimeOnly(DateTime dt, TimeStampPrecision precision)
+        => ToFIXTimeOnly(TimeOnly.FromDateTime(dt), precision);
+
+    private static DateTime TimeOnlyFromNanoString(string str)
+    {
+        return ConvertFromNanoString(str, TIME_ONLY_FORMATS);
+    }
+
+    private static DateTime DateTimeFromNanoString(string str)
+    {
+        return ConvertFromNanoString(str, DATE_TIME_FORMATS);
+    }
+
+    private static DateTime ConvertFromNanoString(string str, string[] formats)
```

Comment:
I don't remember much of this, but I'm pretty sure I deleted this function, was it brought back on purpose?

Reply:
Yes, you did.  I have a copy of your old branch.  I'll get it straightened out.